clients/web/src/utilities/jquery/promise.js

```diff
@@ -0,0 +1,6 @@
+import $ from 'jquery';
+
+// $.when behave differently with zero, one or more argument. This helper method minics Promise.all() behavior
+$.whenAll = function (promises) {
```

Comment:
Let's call this `_whenAll`.

Per @zachmullen 's comment, consider putting it as a general Girder utility, rather than attaching it to `$`..